### PR TITLE
Add workflow setters

### PR DESCRIPTION
**Pull request by niekdejonge**
Solves suggestion of @florian-huber 

Adds setters for workflow attributes back in to prevent changing anything in workflow that will not be passed on the SpectrumProcessor.

### CODE REVIEW

**Review by florian-huber — APPROVED** [review]
Fine with making workflow a hidden attribute! Thanks Niek.

File: matchms/Pipeline.py [hunk 1]
```diff
@@ -377,15 +376,30 @@ def check_files_exist(filenames):
     # Getter & Setters
     @property
     def score_computations(self):
-        return self.workflow.get("score_computations")
+        return self.__workflow.get("score_computations")
```
Comment (hechth) [inline matchms/Pipeline.py]: please add return type annotations

File: matchms/Pipeline.py [hunk 2]
```diff
@@ -377,15 +376,30 @@ def check_files_exist(filenames):
...
+
+    @score_computations.setter
+    def score_computations(self, computations):
+        self.__workflow["score_computations"] = computations
+        check_score_computation(score_computations=self.score_computations)
 
     @property
     def query_filters(self):
```
Comment (hechth) [inline matchms/Pipeline.py]: same here - return types would be very helpful!

File: matchms/Pipeline.py [hunk 3]
```diff
@@ -377,15 +376,30 @@ def check_files_exist(filenames):
...
-        return self.workflow.get("query_filters")
+        return self.__workflow.get("query_filters")
+
+    @query_filters.setter
+    def query_filters(self, filters):
```
Comment (hechth) [inline matchms/Pipeline.py]: type hints for the filters parameter are missing

File: matchms/Pipeline.py [hunk 4]
```diff
@@ -377,15 +376,30 @@ def check_files_exist(filenames):
...
+        self.__workflow["query_filters"] = filters
+        self._initialize_spectrum_processor_queries()
 
     @property
     def reference_filters(self):
-        return self.workflow.get("reference_filters")
+        return self.__workflow.get("reference_filters")
+
+    @reference_filters.setter
+    def reference_filters(self, filters):
```
Comment (hechth) [inline matchms/Pipeline.py]: type hints for the filters parameter are missing

**Pull request merged**